qt/scientific_interfaces/Direct/ALFView_model.cpp

```diff
@@ -0,0 +1,134 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "ALFView_model.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/NumericAxis.h"
+
+#include "MantidGeometry/Instrument.h"
+
+#include "MantidKernel/Unit.h"
+
+#include <utility>
+
+namespace {
+const std::string TMPNAME = "ALF_tmp";
+const std::string INSTRUMENTNAME = "ALF";
+const std::string WSNAME = "ALFData";
+const int ERRORCODE = -999;
+} // namespace
+
+using namespace Mantid::API;
+namespace MantidQt {
+namespace CustomInterfaces {
+
+void ALFView_model::loadEmptyInstrument() {
+  auto alg =
+      Mantid::API::AlgorithmManager::Instance().create("LoadEmptyInstrument");
+  alg->initialize();
+  alg->setProperty("OutputWorkspace", WSNAME);
+  alg->setProperty("InstrumentName", INSTRUMENTNAME);
+  alg->execute();
+}
+/*
+ * Loads data for use in ALFView
+ * Loads data, normalise to current and then converts to d spacing
+ * @param name:: string name for ALF data
+ * @return std::pair<int,std::string>:: the run number and status
+ */
+std::pair<int, std::string> ALFView_model::loadData(const std::string &name) {
```

Comment:
Thanks for the changes to this. I think this better encapsulates the operation :+1:

Reply:
does have the advantage of if other areas want a custom version, we just need to change the model.